pkg/database/database.go

```diff
@@ -1617,6 +1639,13 @@ func (d *db) Close() (err error) {
 
 	d.sqlInit.Wait() // Wait for SQL Engine initialization to conclude
 
+	if d.isDataRetentionEnabled() {
+		d.quitOnce.Do(func() {
```

Comment:
quitOnce might not be needed

Reply:
This is required in case close is called twice, was failing in one test case

Reply:
aham, probably it's better to have a flag in the database level to indicate the database is already closed. So to avoid any action when calling close more than once

Reply:
resolved